doc/refman/RefMan-tac.tex

```diff
@@ -2688,6 +2701,88 @@ \subsection{\tt inversion \ident}
 Note that \texttt{H} has been substituted by \texttt{(LeS n m0 l)} and
 \texttt{m} by \texttt{(S m0)}.
 
+\example{Using \texorpdfstring{\texttt{inversion\_sigma}}{inversion\_sigma}}
+
+Let us consider the following inductive type of length-indexed lists,
+and a lemma about inverting equality of \texttt{cons}:
+
+\begin{coq_eval}
+Reset Initial.
+\end{coq_eval}
+
+\begin{coq_example*}
+Inductive vec A : nat -> Type :=
+| nil : vec A O
+| cons {n} (x : A) (xs : vec A n) : vec A (S n).
+
+Lemma invert_cons : forall A n x xs y ys,
+    @cons A n x xs = @cons A n y ys
+    -> xs = ys.
+Proof.
+intros A n x xs y ys H.
+\end{coq_example*}
+
+\begin{coq_example}
+Show.
+\end{coq_example}
+
+After performing \texttt{inversion}, we are left with an equality of
+\texttt{existT}s:
+
+\begin{coq_example}
+inversion H.
+\end{coq_example}
+
+We can turn this equality into a usable form with
+\texttt{inversion\_sigma}:
+
+\begin{coq_example}
+inversion_sigma.
+\end{coq_example}
+
+To finish cleaning up the proof, we will need to prove that all proofs
+of \texttt{n = n} for \texttt{n} a \texttt{nat} are \texttt{eq\_refl}:
+
+\begin{coq_example*}
+Fixpoint adjust_eq_nat {n m : nat} : n = m -> n = m
+:= match n, m with
+   | O, O => fun _ => eq_refl
+   | S n', S m' => fun p => f_equal S (@adjust_eq_nat n' m' (f_equal pred p))
+   | _, _ => fun p => p
+   end.
+Lemma adjust_eq_nat_refl {n} (p : n = n) : adjust_eq_nat p = eq_refl.
+Proof.
+  induction n as [|n IHn]; [ reflexivity | simpl ].
+  rewrite IHn; reflexivity.
+Qed.
+Lemma adjust_eq_nat_eq {n m} (p : n = m) : adjust_eq_nat p = p.
+Proof. destruct p; apply adjust_eq_nat_refl. Qed.
+Lemma nat_UIP {n : nat} (p : n = n) : p = eq_refl.
+Proof.
+  transitivity (adjust_eq_nat p); [ symmetry; apply adjust_eq_nat_eq | apply adjust_eq_nat_refl ].
+Qed.
```

Comment:
This proof skeleton comes from modifying the encode-decode proof a little.  In any case, though, the proofs are structurally roughly the same, the only differences are that where the Eqdep_dec proof uses `True`, I use `x = x`, and that Eqdep_dec fuses the recursion of my `adjust_eq_nat` with the induction of my `adjust_eq_nat_refl`.  Using `Eqdep_dec.UIP_refl_nat` is simpler, though, so I'll just use that; I didn't do it in the first place because I was too lazy to look up where that proof lived, and figured that it wasn't hard to rewrite it from scratch.